chain/epoch_manager/src/lib.rs

```diff
@@ -666,10 +666,30 @@ impl EpochManager {
         Ok(self.epoch_validators_ordered_unique.cache_get(epoch_id).unwrap())
     }
 
+    /// get_heuristic_block_approvers_ordered: block producers for epoch
+    /// get_all_block_approvers_ordered: block producers for epoch, slashing info, sometimes block producers for next epoch
+    pub fn get_heuristic_block_approvers_ordered(
+        &mut self,
+        epoch_id: &EpochId,
+    ) -> Result<Vec<ApprovalStake>, EpochError> {
+        let epoch_info = self.get_epoch_info(epoch_id)?;
+        let mut settlement = vec![];
+        for validator_id in epoch_info.block_producers_settlement.iter() {
```

Comment:
there is an existing function `get_all_block_producers_settlement`

Reply:
`get_all_block_producers_settlement` takes a block hash, which is not available

Reply:
I see, but we can use the `epoch_validators_ordered` cache